modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSource.java

```diff
@@ -221,15 +222,21 @@ protected void doRun() throws Exception {
                 }
 
                 @Override
-                public void onFailure(Exception t) {
-                    if (t instanceof ResponseException) {
-                        ResponseException re = (ResponseException) t;
+                public void onFailure(Exception e) {
+                    if (e instanceof ResponseException) {
+                        ResponseException re = (ResponseException) e;
                         if (RestStatus.TOO_MANY_REQUESTS.getStatus() == re.getResponse().getStatusLine().getStatusCode()) {
-                            listener.onRetryableFailure(t);
+                            listener.onRetryableFailure(e);
                             return;
                         }
+                        if (RestStatus.UNAUTHORIZED.getStatus() == re.getResponse().getStatusLine().getStatusCode()) {
```

Comment:
No. It'd look like:
* 401 -> 401
* 403 -> 403
* 4xx -> 400
* 5xx -> 500

I don't have an exception I can use that lets my copy the status. I could certainly build a new one but I know we're kind of against building new exceptions?